src/libultra/os/getintmask.s

```diff
@@ -0,0 +1,48 @@
+#include "ultra64/asm.h"
+#include "ultra64/R4300.h"
+#include "ultra64/rcp.h"
+#include "ultra64/exception.h"
+
+.set noat
+.set noreorder
+
+.section .text
+
+.balign 16
+
+LEAF(osGetIntMask)
+    // Extract interrupt enable bits from current SR
+    mfc0    $v0, C0_SR
+    andi    $v0, $v0, (SR_IMASK | SR_IE)
+    // Get value of __OSGlobalIntMask
+    lui     $t0, %hi(__OSGlobalIntMask)
+    addiu   $t0, %lo(__OSGlobalIntMask)
+    lw      $t1, 0($t0)
+    // Bitwise-OR in the disabled CPU bits of __OSGlobalIntMask
+    li      $at, ~0
+    xor     $t0, $t1, $at
+    andi    $t0, $t0, SR_IMASK
+    or      $v0, $v0, $t0
```

Comment:
This code and the other global interrupt mask use below is probably subject to the bug described in `setintmask.s`, might be nice to mention it

Reply:
I copied the comments from setintmask.s and lightly modified them, does it look correct?

Reply:
It looks right, though I'd just redirect to the setintmask comment for the bulk of the bug explanation rather than reproduce it here, something like
```
 * @bug Some usage of the global interrupt mask is broken both in the set/get interrupt mask routines
 *       and in the exception handler routines. See the comment for osSetIntMask for details.
```

Reply:
Makes sense, done